app/views/shared/_spinner-button.html.erb

```diff
@@ -0,0 +1,19 @@
+<%#
+yields: Button or link markup (required).
+%>
+<%
+  raise "no block content given" unless yield.presence
+%>
+<div class="spinner-button">
+  <div class="spinner-button__content">
+    <%= yield %>
```

Comment:
```suggestion
    <%= yield.presence ||  raise "no block content given" %>
```

What if we removed line 5 above and did this?


In "normal" ruby, `yield` immediately evaluates the block given, so if that's the same here, it would result in calling whatever block twice, which might have unwanted side effects.

This may not be normal Ruby so I'm not sure if the result if the `yield` gets cached, or if we care... but for reference `yield.presence` to check for a result wouldn't work in normal Ruby if there was no block supplied, calling `yield` would result in a `LocalJumpError`...

```ruby
require 'active_support/core_ext/object/blank'
def test
  yield.presence
end

test { 1 }
# => 1
test
# => LocalJumpError (no block given (yield))
```
```

Reply:
> What if we removed line 5 above and did this?

Yeah, that seems fine and reasonable given your rationale. For context, the thinking here was to follow a pattern where the first few lines of a "component" would initialize (normalize, default, validate) locals used in the partial, thus making it predictable to know where to find this logic. [Alert](https://github.com/18F/identity-idp/blob/60a220cd4b6ad305dc7265c21d833c59f7c7d41d/app/views/shared/_alert.html.erb#L1-L20) being an example where it's a bit more involved.

Reply:
That seems reasonable, we could make that line something like

```
content = yield.presence || raise "no block content given"
```

Reply:
> That seems reasonable, we could make that line something like
> 
> ```
> content = yield.presence || raise "no block content given"
> ```

Updated with minor revision in ff11a1f (and I [learned a thing or two about operator precedence](https://avdi.codes/how-to-use-rubys-english-andor-operators-without-going-nuts/) in the process!).